### PR TITLE
feat(delete_customer,delete_plan): Update documentation

### PR DESCRIPTION
**Pull request by vincent-pochet**
## Context

Currently, there is no way to delete objects linked to a subscription. This can be a real limitation during PoCs and onboarding, as users have to ping us to delete plans and customers.

The goal of this feature is to be able to soft-delete a plan or a customer linked to an active or terminated subscription.

## Description

The goal of this PR is to:
- add two counters in the plan response: `active_subscription_count` and `draft_invoice_count`
- add the documentation of the customer delete route
- remove the HTTP 405 errors on delete customer and delete coupons actions

## Relation

Closes:
- https://github.com/getlago/lago-docs/pull/111
- https://github.com/getlago/lago-docs/pull/116
- https://github.com/getlago/lago-docs/pull/117

### CODE REVIEW

File: docs/api/03_plans/plan-object.mdx [hunk 1]
```diff
@@ -122,6 +124,8 @@ This object represents a plan.<br></br>
 | **trial_period** &nbsp &nbsp <Type>Float</Type> | Number of days for free trial. |
 | **pay_in_advance** &nbsp &nbsp <Type>Boolean</Type> | Value should be `true` for payment in advance and `false` for payment in arrear. |
 | **bill_charges_monthly** &nbsp &nbsp <Type>Boolean</Type> | It should be set to `true` if monthly billing is needed on yearly plan. |
+| **active_subscriptions_count** &nbsp &nbsp <Type>Integer</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Count of active subscription attached to the plan. This field can be used to know the impact of deleting this plan. |
```
Comment (rsempe) [inline docs/api/03_plans/plan-object.mdx:127]:
```suggestion
| **active_subscriptions_count** &nbsp &nbsp <Type>Integer</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Count of active subscriptions attached to the plan. This field can be used to know the impact of deleting this plan. |
```

File: docs/api/04_customers/destroy-customer.mdx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            // Configure HTTP bearer authorization: bearerAuth
+            Configuration.Default.AccessToken = "YOUR_ACCESS_TOKEN";
+
+            var apiInstance = new CustomersApi(Configuration.Default);
+            var externalID = "external_id";  // string | External id of the existing customer
+
+            try
+            {
+                // Delete a customer
+                Coupon result = apiInstance.DestroyCustomer(externalID);
```
Comment (rsempe) [inline docs/api/04_customers/destroy-customer.mdx:109]:
```suggestion
                Customer result = apiInstance.DestroyCustomer(externalID);
```

File: docs/api/04_customers/destroy-customer.mdx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        SetApiKey("__YOUR_API_KEY__")
+
+      customer, err := lagoClient.Customer().Delete("external_id")
+      if err != nil {
+        // Error is *lago.Error
+        panic(err)
+      }
+
+      // coupon is *lago.Coupon
+      fmt.Println(coupon)
```
Comment (rsempe) [inline docs/api/04_customers/destroy-customer.mdx:78]:
```suggestion
      // customer is *lago.Customer
      fmt.Println(customer)
```

File: docs/api/04_customers/destroy-customer.mdx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  ```csharp
+using System.Collections.Generic;
+using System.Diagnostics;
+using Org.OpenAPITools.Api;
+using Org.OpenAPITools.Client;
+using Org.OpenAPITools.Model;
+
+namespace Example
+{
+    public class DestroyCouponExample
```
Comment (rsempe) [inline docs/api/04_customers/destroy-customer.mdx:95]:
```suggestion
    public class DestroyCustomerExample
```

File: docs/guide/02_self-hosting/tracking.md [hunk 5]
```diff
@@ -35,6 +35,8 @@ This event flow enables Lago to understand the full activation flow, from the re
 - `invoice_created`: when a new invoice is emitted for a specific customer
 - `payment_status_changed`: when the payment status of an invoice changes
 - `credit_note_issued`: when a new credit note is issued for a customer
+- `billable_metric_deleted`: when a billable metric is deleted
```
Comment (rsempe) [inline docs/guide/02_self-hosting/tracking.md:38]: 👏

File: docs/guide/02_self-hosting/tracking.md [hunk 6]
```diff
@@ -35,6 +35,8 @@ This event flow enables Lago to understand the full activation flow, from the re
...
+- `plan_deleted`: when a plan is deleted
```
Comment (rsempe) [inline docs/guide/02_self-hosting/tracking.md:39]: to not forget `customer_deleted` as well

**Review by rsempe — APPROVED** [review]

File: package.json [hunk 7]
```diff
@@ -22,7 +22,8 @@
     "clsx": "^1.1.1",
     "prism-react-renderer": "^1.3.1",
     "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react-dom": "^17.0.2",
+    "yarn": "^1.22.19"
```
Comment (ansmonjol) [inline package.json:26]: is that intended ?

**Review by ansmonjol — APPROVED** [review]

**Pull request merged**